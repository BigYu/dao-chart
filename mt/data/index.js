module.exports = async function(ctx, next) {
  const { name, parameters } = ctx.request.body;
  const handle = require(`./${name}-handler.js`);

  ctx.body = await handle(parameters);
  await next();
};
