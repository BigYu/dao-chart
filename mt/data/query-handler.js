module.exports = function(parameters) {
  return { parameters };
};
