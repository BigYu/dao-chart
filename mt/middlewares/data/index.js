const Promise = require('bluebird');
const Boom = require('boom');
const _ = require('lodash');

async function handleSingleQuery(query) {
  const { name, parameters } = query;

  if (_.isEmpty(name) || !_.isString(name)) {
    throw Boom.badRequest('Missing "name"');
  }

  if (_.isEmpty(parameters) || !_.isObject(parameters)) {
    throw Boom.badRequest('Missing "parameters"');
  }

  const handle = require(`./handlers/${name}.js`);

  return handle(parameters);
}

module.exports = () => async function(ctx, next) {
  const { body } = ctx.request;

  if (_.isArray(body)) {
    ctx.body = await Promise.map(body, async q => handleSingleQuery(q));
    await next();
  } else {
    ctx.body = await handleSingleQuery(ctx.request.body);
    await next();
  }
};
