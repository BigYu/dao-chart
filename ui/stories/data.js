export const source = [
  ['product', '2015', '2016', '2017'],
  ['Matcha Latte', 43.3, 85.8, 93.7],
  ['Milk Tea', 83.1, 73.4, 55.1],
  ['Cheese Cocoa', 86.4, 65.2, 82.5],
  ['Walnut Brownie', 72.4, 53.9, 39.1],
];

export const pieSource = [
  ['name', 'value'],
  ['直接访问', 335],
  ['邮件营销', 310],
  ['联盟广告', 234],
  ['视频广告', 135],
  ['搜索引擎', 1548],
];
