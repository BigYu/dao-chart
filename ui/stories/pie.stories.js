import React from 'react';
import { storiesOf } from '@storybook/react';
import Pie from '../src/component/pie';
import { pieSource } from './data';


storiesOf('Pie Chart', module)
  .add('pie chart - percent', () => (
    <Pie percent={28} />
  ))
  .add('pie chart - data', () => (
    <Pie source={pieSource} />
  ))
  .add('pie chart - data - withoutLegend', () => (
    <Pie hasLegend={false} source={pieSource} />
  ));
