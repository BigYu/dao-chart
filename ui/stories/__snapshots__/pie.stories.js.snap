// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of pie chart 1`] = `
<EchartsReact
  className=""
  echarts={Object {}}
  lazyUpdate={false}
  loadingOption={null}
  notMerge={false}
  onChartReady={[Function]}
  onEvents={Object {}}
  option={
    Object {
      "legend": Object {
        "data": Array [
          Object {
            "icon": "circle",
            "name": "直接访问",
          },
          Object {
            "icon": "circle",
            "name": "邮件营销",
          },
          Object {
            "icon": "circle",
            "name": "联盟广告",
          },
          Object {
            "icon": "circle",
            "name": "视频广告",
          },
          Object {
            "icon": "circle",
            "name": "搜索引擎",
          },
        ],
        "formatter": [Function],
        "orient": "vertical",
        "right": "0",
        "top": "middle",
      },
      "series": Array [
        Object {
          "data": Array [
            Object {
              "name": "直接访问",
              "value": 335,
            },
            Object {
              "name": "邮件营销",
              "value": 310,
            },
            Object {
              "name": "联盟广告",
              "value": 234,
            },
            Object {
              "name": "视频广告",
              "value": 135,
            },
            Object {
              "name": "搜索引擎",
              "value": 1548,
            },
          ],
          "hoverOffset": 0,
          "label": Object {
            "normal": Object {
              "show": false,
            },
          },
          "labelLine": Object {
            "normal": Object {
              "show": false,
            },
          },
          "left": "center",
          "top": "center",
          "type": "pie",
        },
      ],
      "tooltip": Object {
        "formatter": "{b}: {d}%",
        "trigger": "item",
      },
    }
  }
  opts={Object {}}
  showLoading={false}
  style={Object {}}
  theme={null}
/>
`;
